routers/api/v1/api.go

```diff
@@ -1195,6 +1195,7 @@ func Routes() *web.Router {
 					m.Get("/*", repo.GetBranch)
 					m.Delete("/*", reqToken(), reqRepoWriter(unit.TypeCode), mustNotBeArchived, repo.DeleteBranch)
 					m.Post("", reqToken(), reqRepoWriter(unit.TypeCode), mustNotBeArchived, bind(api.CreateBranchRepoOption{}), repo.CreateBranch)
+					m.Post("/rename", tokenRequiresScopes(auth_model.AccessTokenScopeCategoryRepository), reqRepoWriter(unit.TypeCode), bind(api.RenameBranchRepoOption{}), repo.RenameBranch)
```

Comment:
Two questions here:

1. If a branch is named `rename`, then we can get it by `GET /branches/rename`, but `POST /branches/rename` seems strange (REST-ful is painful in many cases, quite unclear)
2. Why `tokenRequiresScopes` here? I do not see it in other routes like the Delete and Post above.

----

For the REST-ful design problem:
* I think GitHub implemented that API in early days without carful design and maybe we do not need to follow it (and it seems that our framework doesn't support it well).
* `POST /branches/rename` seems quite ambiguous
* Usually "PATCH" method means "change something"

So maybe the following design makes sense?

```
m.Delete("/*", reqToken(), reqRepoWriter(unit.TypeCode), mustNotBeArchived, repo.DeleteBranch)
m.Patch("/*", reqToken(), reqRepoWriter(unit.TypeCode), mustNotBeArchived, repo.UpdateBranch)
```

(read "current branch" from URL path and read other "update options" from body)

Reply:
> 1. If a branch is named rename, then we can get it by GET /branches/rename, but POST /branches/rename (other branch) seems strange (REST-ful is painful in many cases, quite unclear)

Yes, I do agree that this edge case makes this endpoint seem awkward.

> 2. Why tokenRequiresScopes here? I do not see it in other routes like the Delete and Post above.

This might have been a relic from the previous PR and I just didn't remove it. Since the other methods don't enforce such a scope I'll remove it.

> For the REST-ful design problem:

> I think GitHub implemented that API in early days without carful design and maybe we do not need to follow it (and it seems that our framework doesn't support it well).
POST /branches/rename seems quite ambiguous
Usually "PATCH" method means "change something"
So maybe the following design makes sense?

> m.Delete("/*", reqToken(), reqRepoWriter(unit.TypeCode), mustNotBeArchived, repo.DeleteBranch)
> m.Patch("/*", reqToken(), reqRepoWriter(unit.TypeCode), mustNotBeArchived, repo.UpdateBranch)
(read "current branch" from URL path and read other "update options" from body)

Given the problems we've encountered thus far trying to work with GitHub's design, I think this is a better choice.

I was wondering if I needed to include additional branch-related fields that could be updated, but it looks like it's just the name (branch protection is already covered in its own endpoint). I'll make the necessary changes